src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/service/GlobalGroupService.java

```diff
@@ -27,34 +27,91 @@
 
 import lombok.RequiredArgsConstructor;
 import org.eclipse.jetty.util.StringUtil;
-import org.niis.xroad.centralserver.openapi.model.GlobalGroup;
+import org.niis.xroad.centralserver.openapi.model.GlobalGroupResource;
 import org.niis.xroad.centralserver.restapi.converter.GlobalGroupConverter;
+import org.niis.xroad.centralserver.restapi.dto.GlobalGroupUpdateDto;
+import org.niis.xroad.centralserver.restapi.entity.GlobalGroup;
 import org.niis.xroad.centralserver.restapi.entity.GlobalGroupMember;
+import org.niis.xroad.centralserver.restapi.entity.SystemParameter;
+import org.niis.xroad.centralserver.restapi.openapi.InternalServerErrorException;
 import org.niis.xroad.centralserver.restapi.repository.GlobalGroupRepository;
+import org.niis.xroad.centralserver.restapi.repository.SystemParameterRepository;
+import org.niis.xroad.restapi.config.audit.AuditDataHelper;
+import org.niis.xroad.restapi.openapi.BadRequestException;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import static org.niis.xroad.centralserver.restapi.service.SystemParameterService.SECURITY_SERVER_OWNERS_GROUP;
+import static org.niis.xroad.restapi.config.audit.RestApiAuditProperty.CODE;
+import static org.niis.xroad.restapi.config.audit.RestApiAuditProperty.DESCRIPTION;
+
 @Service
 @Transactional
 @RequiredArgsConstructor
 public class GlobalGroupService {
 
+    private final AuditDataHelper auditDataHelper;
     private final GlobalGroupRepository globalGroupRepository;
     private final GlobalGroupConverter globalGroupConverter;
+    private final SystemParameterRepository systemParameterRepository;
 
-    public Set<GlobalGroup> findGlobalGroups(String containsMember) {
+    public Set<GlobalGroupResource> findGlobalGroups(String containsMember) {
         return globalGroupRepository.findAll().stream()
                 .filter(globalGroup -> isMemberExistsInGlobalGroup(containsMember, globalGroup.getGlobalGroupMembers()))
                 .map(globalGroupConverter::convert)
                 .collect(Collectors.toSet());
     }
 
+    public GlobalGroupResource getGlobalGroup(Integer groupId) {
+        return globalGroupRepository.findById(groupId)
+                .map(globalGroupConverter::convert)
+                .orElseThrow(() -> new InternalServerErrorException("Failed to retrieve global group"));
+    }
+
+    public void deleteGlobalGroup(Integer groupId) {
+        globalGroupRepository.findById(groupId).ifPresent(this::handleInternalDelete);
```

Comment:
Currently as it is when the `GlobalGroup` with the given id does not exist, then the ifPresent block is not executed and the server responds with HTTP 200. From the standpoint of REST API design I would prefer to respond with HTTP 404 (even when dealing with delete operation)